crates/anemo/src/config.rs

```diff
@@ -45,12 +45,39 @@ pub struct Config {
     #[serde(skip_serializing_if = "Option::is_none")]
     pub connection_backoff_ms: Option<u64>,
 
+    /// Set a timeout, in milliseconds, for all inbound and outbound connects.
+    ///
+    /// In unspecified, this will default to `10,000` milliseconds.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub connect_timeout_ms: Option<u64>,
+
     /// Maximum number of concurrent connections to attempt to establish at a given point in time.
     ///
     /// If unspecified, this will default to `100`.
     #[serde(skip_serializing_if = "Option::is_none")]
     pub max_concurrent_outstanding_connecting_connections: Option<usize>,
 
+    /// Maximum number of concurrent connections to have established at a given point in time.
+    ///
+    /// This limit is applied in the following ways:
+    ///  - Inbound connections from [`KnownPeers`] with [`PeerAffinity::High`] bypass this limit. All
```

Comment:
small request - Can you reword this to make it clearer whether the "limit bypass" cases count towards the limit, even if they bypass it? (I think yes but not 100% sure)